Mathlib/RingTheory/MvPowerSeries/Trunc.lean

```diff
@@ -105,7 +110,90 @@ theorem trunc_map [CommSemiring S] (n : σ →₀ ℕ) (f : R →+* S) (p : MvPo
     trunc S n (map σ f p) = MvPolynomial.map f (trunc R n p) := by
   ext m; simp [coeff_trunc, MvPolynomial.coeff_map, apply_ite f]
 
-end Trunc
+end TruncLT
+
+section TruncLE
+
+variable [CommSemiring R] (n : σ →₀ ℕ)
+
+/-- Auxiliary definition for the truncation function. -/
+def truncFun' (φ : MvPowerSeries σ R) : MvPolynomial σ R :=
+  ∑ m ∈ Finset.Iic n, MvPolynomial.monomial m (coeff R m φ)
+
+/-- Coefficients of the truncated function. -/
+theorem coeff_truncFun' (m : σ →₀ ℕ) (φ : MvPowerSeries σ R) :
+    (truncFun' n φ).coeff m = if m ≤ n then coeff R m φ else 0 := by
+  classical
+  simp [truncFun', MvPolynomial.coeff_sum]
+
+variable (R)
+
+/-- The `n`th truncation of a multivariate formal power series to a multivariate polynomial -/
```

Comment:
```suggestion
/-- 
The `n`th truncation of a multivariate formal power series to a multivariate polynomial.

If `f : MvPowerSeries σ R` and `n : σ →₀ ℕ` is a (finitely-supported) function from `σ` to the naturals, then `trunc' R n f` is the multivariable power series obtained from `f` by keeping only the monomials $c\prod_i X_i^{a_i}$ where `a i ≤ n i` for all `i`.
-/
```
In my opinion the docstring for `trunc` is also lousy (it is far too imprecise) and should be clarified. But apart from that the PR looks good to me.

Reply:
I added a lot of docstrings.